### ISSUE TITLE
[Rule Request] Prohibited Interface Builder

### ISSUE THREAD

**Issue by marcelofabri** [realm/SwiftLint#2365]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Rule Request

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

Using Interface Builder (storyboards or nibs) is a constant subject of heated debates in the iOS community. Some projects don't use it at all. For those cases, a rule validating the use of `@IBAction` and `@IBOutlet` would be helpful.

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

```swift
// would trigger
@IBOutlet var label: UILabel!

@IBAction func buttonTapped(_ sender: UIButton) {}

// wouldn't trigger
var label: UILabel!

@objc func buttonTapped(_ sender: UIButton) {}
```

3. Should the rule be configurable, if so what parameters should be configurable?

Just severity.

4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a
   rule as opt-in.

Opt-in, as using IB or not is a project decision.
